sdk/core/Microsoft.Azure.Core.NewtonsoftJson/src/NewtonsoftJsonObjectSerializer.cs

```diff
@@ -68,7 +68,7 @@ public override object Deserialize(Stream stream, Type returnType, CancellationT
             Argument.AssertNotNull(returnType, nameof(returnType));
 
             using StreamReader reader = new StreamReader(stream, UTF8NoBOM, true, DefaultBufferSize, true);
-            return _serializer.Deserialize(reader, returnType);
+            return _serializer.Deserialize(reader, returnType)!;
```

Comment:
Is this guaranteed to be null, or should the method return `object?` ?

Reply:
if we change nullability on the public api I think thats a break?  basically I am just keeping the old behavior, if it was null before it will do the same thing as a null now.